### ISSUE TITLE
Support token exchange in Google connector

### ISSUE THREAD

**Issue by rorynickolls-skyral** [dexidp/dex#3768]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/dexidp/dex/blob/master/.github/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/dexidp/dex/issues) for an issue that matches the one I want to file, without success.

### Problem Description

We are performing machine authentication by exchanging a Google access token (from an impersonated service account) for a dex-issued JWT, as in [these docs](https://dexidp.io/docs/guides/token-exchange/).
 
The Google connector does not appear to support token exchange, but the OIDC connector configured with Google does. 

The limitation of using the OIDC connector is that we cannot get group claims; it appears that the Google connector handles those as a special case.

### Proposed Solution

Support token exchange in the Google connector. I think it should be possible given that token exchange is possible with Google via the OIDC connector, and would enable group claims when doing so.

**Comment by nabokihms** [comment]
Yes, sounds fine to me. It should be similar to the oidc implementation.

**Comment by rorynickolls-skyral** [comment]
I've been able to get this working locally by implementing token exchange in the same way as the OIDC connector like you suggested.

The result is that I can perform token exchange on the Google connector using a service account access token, and the resulting JWT includes group claims alongside the usual claims.

I'll aim to get it tidied up and tested properly in the New Year.
